### ISSUE TITLE
EnumConverter for enum *arrays*

### ISSUE THREAD

**Issue by gkresic** [jOOQ/jOOQ#13982]
### Use case

It would be great if jOOQ's `org.jooq.impl.EnumConverter` could be used/configured for converting columns representing arrays of enum UDTs.

Currently, I'm using `forcedType` declarations for converting columns of any enum type (snippet from Gradle config):

```
forcedType {
    types = '(my_enum|MY_ENUM)'
    userType = 'com.foo.MyEnum'
    enumConverter = true
}
```

This works, as expected.

However, if I try to do the same with arrays (note the underscore - standard PostgreSQL prefix for arrays of certain type):

```
forcedType {
    types = '_(my_enum|MY_ENUM)'
    userType = 'com.foo.MyEnum'
    enumConverter = true
}
```

In this case, jOOQ code generator produces code that doesn't compile.

Could `org.jooq.ArrayConverter` be configured somehow?

### jOOQ Version

jOOQ Open Source 3.17.4

### Database product and version

PostgreSQL 14.4

### Java Version

OpenJDK 17

### OS Version

Ubuntu 22.04

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.3.6

**Comment by lukaseder** [comment]
Thanks a lot for your feature request. As mentioned on the user group, I think it does make sense to make the `enumConverter` flag array aware:
https://groups.google.com/g/jooq-user/c/wvp_laEoPAo

This could be particularly interesting in the edge case of using multi dimensional enum arrays: https://github.com/jOOQ/jOOQ/issues/252

The workaround here is to configure the array converter manually, by calling creating the base type's enum converter and then calling `Converter.forArrays()` on it.

**Comment by lukaseder** [comment]
It seems that for now, this feature will be available in PostgreSQL only, as in H2, enum arrays don't work as expected, yet: https://github.com/h2database/h2database/issues/3632

**Comment by gkresic** [comment]
I'm not sure how related to this feature request this is, but it seems jOOQ doesn't recognize enum array types defined in custom schema. Interestingly, regular enum types are recognized.

For example, this works (`example` is a schema name, `foo` is enum):

```
forcedType {
	includeTypes = '(example.foo)'
	userType = 'com.example.Foo'
	enumConverter = true
}
```

However, this (note the underscore in type name, what is how PostgreSQL defines array types):

```
forcedType {
	includeTypes = '(example._foo)'
	userType = 'com.example.Foo[]'
	enumConverter = true
}
```

Throws an error during code generation:

```
Unused forced types      : There are unused forced types, which have not been used by this generation run.
This can be because of misconfigurations, such as, for example:
- case sensitive regular expressions
- regular expressions depending on whitespace (Pattern.COMMENTS is turned on!)
- missing or inadequate object qualification
- the object to which the configuration was applied in the past has been dropped
Try turning on DEBUG logging (-X in Maven, --debug in Gradle, and <logging/> in jOOQ) to get additional info about the schema
To turn off individual warnings, specify <ignoreUnused>true</ignoreUnused> on the relevant object(s).
Unused forced type       : <priority>0</priority><userType>com.example.Foo[]</userType><hidden>false</hidden><enumConverter>true</enumConverter><includeTypes>(example._foo)</includeTypes><nullability>ALL</nullability><objectType>ALL</objectType><ignoreUnused>false</ignoreUnused>
```

**Comment by lukaseder** [comment]
My guess is that we don't yet include any schema qualification for user defined types in that `includeTypes` flag. Have you tried removing the qualification? If that's it, please can you create a new issue?

**Comment by gkresic** [comment]
> My guess is that we don't yet include any schema qualification for user defined types in that includeTypes flag.

Nope, that doesn't seem to be the case - note that a user type defined in a custom schema **works**, but only for scalar types and not for the arrays.

This **does** work (`example` is a custom schema name):
```
forcedType {
	includeTypes = '(example.foo)'
	userType = 'com.example.Foo'
	enumConverter = true
}
```

My guess is that code generator searches for `_` as a **first** character in type definition as a discriminator for arrays. And although that's true when working without schema prefix (`_foo`), it is not when you prepend a schema in front, in which case it comes after schema identifier (`schema._foo`).

**Comment by lukaseder** [comment]
OK, could you create an issue for that then? It's a different problem from this feature request.

**Comment by gkresic** [comment]
Created #19617.
